tests/integ_tests/workflow_tests/appdata_propagation.rs

```diff
@@ -0,0 +1,63 @@
+use assert_matches::assert_matches;
+use std::time::Duration;
+use temporal_client::{WfClientExt, WorkflowExecutionResult, WorkflowOptions};
+use temporal_sdk::{ActContext, ActivityOptions, WfContext, WorkflowResult};
+use temporal_sdk_core_protos::coresdk::AsJsonPayloadExt;
+use temporal_sdk_core_test_utils::CoreWfStarter;
+
+const TEST_APPDATA_MESSAGE: &str = "custom app data, yay";
+
+struct Data {
+    message: String,
+}
+
+pub async fn appdata_activity_wf(ctx: WfContext) -> WorkflowResult<()> {
+    let data = ctx.app_data::<Data>().expect("appdata exists. qed");
```

Comment:
I'm worried that users will use this to access environment specific data resulting in non-deterministic behavior.
For activities this API is totally fine.

Reply:
I mulled over this quite a bit too.  It is a little like providing nulls in a programming language and trusting everyone will check it all the time (or be cognizant).  However, I ultimately added it in because I imagine someone will ask for this kind of context propagation for things like tracing (which is in one of the go sdk samples).  I'm happy to take it out for now.  Just having it for activities is enough for my own use case.

Reply:
I think we should leave it out for now. Keep in mind the Workflow APIs especially are nowhere close to what I expect them to be later.

Ultimately I think workflows will probably look more like implementing a trait in the long run, in which case you could just include this stuff in the fields of your struct which implements that trait.

Reply:
:+1: For data on `self`